rust-runtime/aws-smithy-http-server/src/runtime_error.rs

```diff
@@ -54,6 +57,12 @@ pub struct RuntimeError {
     pub kind: RuntimeErrorKind,
 }
 
+impl<P> IntoResponse<P> for RuntimeError {
```

Comment:
Nice. Is there any blocker now to making `RuntimeError` generic over the protocol, and implement protocol-specific implementations of `IntoResponse` for them? It's not blocked on moving to the new builder right?

Reply:
After this is merged I think we could replace

```rust
#[derive(Debug)]
pub struct RuntimeError {
    pub protocol: Protocol,
    pub kind: RuntimeErrorKind,
}

impl<P> IntoResponse<P> for RuntimeError {
    fn into_response(self) -> http::Response<crate::body::BoxBody> {
        self.into_response()
    }
}

impl RuntimeError {
    pub fn into_response(self) -> Response {
        /* runtime match on enum */
```

with

```rust
#[derive(Debug)]
pub struct RuntimeError {
    pub kind: RuntimeErrorKind,
}

impl IntoResponse<RestJson1> for RuntimeError {
    fn into_response(self) -> http::Response<crate::body::BoxBody> {
        /* protocol specific implementation */
    }
}
```

I think this should still be superseded by https://github.com/awslabs/smithy-rs/issues/1703 where we do away with `RuntimeError` completely and have protocol specific enums and impls.